### PR TITLE
Port detect_clipping from Solar Forecast Arbiter

### PR DESCRIPTION
**Pull request by wfvining**
Detects inverter clipping based on levels in the data.

## Changes from SFA
* `detect_clipping` renamed to `features.clipping_levels`
* remove `@mask_flags` decorator
* rewrote tests to avoid depending on pvlib to generate data.

### CODE REVIEW

File: pvanalytics/features.py [hunk 1]
```diff
@@ -1 +1,74 @@
...
+def clipping_levels(ac_power, window=4, fraction_in_window=0.75,
+                    rtol=5e-3, levels=2):
+    """Label clipping in AC power data based on levels in the data.
+
+    Parameters
+    ----------
+    ac_power : Series
+        Time series of AC power measurements.
+    window : int
+        Size of the window used to detect clipping.
```
Comment (cwhanse) [inline pvanalytics/features.py:49]:
```suggestion
        Number of data points in a window used to detect clipping.
```

File: pvanalytics/tests/test_features.py [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+import pytest
+import pandas as pd
+from pandas.util.testing import assert_series_equal
+import numpy as np
+from pvanalytics import features
+
+
+@pytest.fixture
+def quadratic():
+    """A downward facing quadratic."""
```
Comment (cwhanse) [inline pvanalytics/tests/test_features.py:11]:
```suggestion
    """A downward facing quadratic with vertex at (0, 1000) and roots at +/-1."""
```

File: pvanalytics/tests/test_features.py [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+    q = -1000 * (np.linspace(-1, 1, 60) ** 2) + 1000
+    return pd.Series(q)
+
+
+@pytest.fixture
+def quadratic_clipped(quadratic):
+    """A downward facing quadratic with clipping"""
```
Comment (cwhanse) [inline pvanalytics/tests/test_features.py:18]:
```suggestion
    """A downward facing quadratic with clipping at y=800"""
```

File: pvanalytics/tests/test_features.py [hunk 4]
```diff
@@ -8,7 +8,11 @@
 
 @pytest.fixture
 def quadratic():
-    """Downward facing quadratic with vertex at (0, 1000) and roots at +/-1."""
+    """Downward facing quadratic.
+
+    Vertex at (30.5, 1000) and roots at 0, 59.
```
Comment (cwhanse) [inline pvanalytics/tests/test_features.py:13]: `0, 59` is the index where the roots are located?  +/-1 are the x-values.

Comment (wfvining) [reply pvanalytics/tests/test_features.py:13]: I'm thinking about it as the index of the Series being the x-values. Having the documentation in those terms makes it easier (for me) to write additional tests that use the fixture.

Comment (cwhanse) [reply pvanalytics/tests/test_features.py:13]: Makes sense. Maybe "Downward facing quadratic with vertex at index position 30" - and redefine `q` as in next comment

File: pvanalytics/tests/test_features.py [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+
+
+@pytest.fixture
+def quadratic():
+    """Downward facing quadratic.
+
+    Vertex at (30.5, 1000) and roots at 0, 59.
+
+    """
+    q = -1000 * (np.linspace(-1, 1, 60) ** 2) + 1000
```
Comment (cwhanse) [inline pvanalytics/tests/test_features.py:16]:
```suggestion
    q = -1000 * (np.linspace(-1, 1, 61) ** 2) + 1000
```

File: pvanalytics/tests/test_features.py (hunk 4 above)
Comment (wfvining) [reply pvanalytics/tests/test_features.py:13]:
```suggestion
    Vertex at index 30 and roots at indices 0, 60.
```

**Pull request merged**